test/src/e2e_vm_tests/test_programs/should_fail/shadowing/shadowed_glob_imports/src/main.sw

```diff
@@ -0,0 +1,39 @@
+script;
+
+dep lib;
+
+// Glob import should not result in any shadowing issues 
+use lib::*;
+
+// const shadowing an imported const with glob 
+const X1 = 7;
+
+// types and traits shadowing imported items with glob 
+struct MyStruct11 {}
+enum MyStruct21 {}
+trait MyStruct31 {}
+abi MyStruct41 {}
+
+
+struct MyEnum12 {}
+enum MyEnum22 {}
+trait MyEnum32 {}
+abi MyEnum42 {}
+
+
+struct MyTrait13 {}
+enum MyTrait23 {}
+trait MyTrait33 {}
+abi MyTrait43 {}
+
+
+struct MyAbi14 {}
+enum MyAbi24 {}
+trait MyAbi34 {}
+abi MyAbi44 {}
+
+
+fn main() {
+    // var shadowing an imported const with glob
+    let X2 = 4;
+}
```

Comment:
It looks like this example passes, but is under the `should_fail` test_programs. Should we create another `shadowing` dir under the `should_pass` directory for those expected to pass?

Reply:
I can do that, but honestly the distinction between `should_pass` and `should_fail` is not very relevant anymore and we should just merge them. I think there's an issue tracking that.